nestkernel/universal_data_logger_impl.h

```diff
@@ -117,11 +117,16 @@ nest::DynamicUniversalDataLogger< HostNode >::DataLogger_::init()
   // +1 because the division result is rounded down.
   next_rec_step_ = ( kernel().simulation_manager.get_time().get_steps() / rec_int_steps_ + 1 ) * rec_int_steps_ - 1;
 
-  // if offset is not 0, adjust next recording step to account for it by
-  // going one interval step back and adding the offset
+  // If offset is not 0, adjust next recording step to account for it by
+  // first setting next recording step to be offset and then iterating until
+  // the variable is greater than current simulation time.
   if ( recording_offset_.get_steps() != 0 )
   {
-    next_rec_step_ = next_rec_step_ - rec_int_steps_ + recording_offset_.get_steps();
+    next_rec_step_ = recording_offset_.get_steps() - 1; // shifted one to left
+    while ( next_rec_step_ <= kernel().simulation_manager.get_time().get_steps() )
```

Comment:
Do we need to loop here? Couldn't we just calculate the correct next_rec_step_?

Reply:
@heplesser I tried for a really long time to calculate the correct `next_rec_step_`, but I never managed to find a calculation that fit for all the different outcomes. If you have a suggestion I could try that of course.

Reply:
I think it's possible to calculate it in one step for all cases, with something along the lines:
```
next_rec_step_ = recording_offset_.get_steps() - 1; // shifted one to left
tmp_next_step_ = ceil( ( kernel().simulation_manager.get_time().get_steps() - next_rec_step_) /  rec_int_steps_ ) * rec_int_steps_;
if ( tmp_next_step_ > 0. )  // needed as tmp_next_step_ can be negative, in which case no update is required
{
  next_rec_step_ += tmp_next_step_;
}
```
using `ceil()` in case the division is rounded down..

Reply:
@zbarni unfortunately the calculation does not work. Test 2 in [test_multimeter_offset](https://github.com/nest/nest-simulator/pull/1564/files#diff-de974299a71b7711c22611787ba866a0R90) get an assert error. I don't really understand why. Test 5 also fails, and this is because it records at wrong times. The test creates a multimeter with `interval=3, offset = 1` and `start=10`, and then it simulates for `10` ms. Then, it creates a second multimeter, with the same `interval, offset` and `start`, and then simulates for 20 ms. The second multimeter should have a `next_rec_time_` of `129`, because it should start recording at `100` and interval is `30` and offset is `10`, but with the new calculation `next_rec_time_` is `99`, and then the multimeter will start recording before it should. I guess something similar is happening with test 2.
When I tried calculating, I often managed to get the correct `next_rec_time_` for some of the cases, but never for all.